autopush/db.py

```diff
@@ -995,19 +1036,23 @@ class DatabaseManager(object):
 
     router = attrib(default=None)                   # type: Optional[Router]
     message_tables = attrib(default=Factory(list))  # type: List[str]
-    current_msg_month = attrib(init=False)          # type: Optional[str]
+    current_msg_month = attrib(default=None,
+                               init=False)          # type: Optional[str]
     current_month = attrib(init=False)              # type: Optional[int]
     _message = attrib(default=None)                 # type: Optional[Message]
+    allow_table_rotation = attrib(default=True)     # type: Optional[bool]
     # for testing:
 
     def __attrs_post_init__(self):
         """Initialize sane defaults"""
-        today = datetime.date.today()
-        self.current_month = today.month
-        self.current_msg_month = make_rotating_tablename(
-            self._message_conf.tablename,
-            date=today
-        )
+        if self.allow_table_rotation:
+            today = datetime.date.today()
+            self.current_month = today.month
```

Comment:
If this is only assigned when allow_table_rotation is True, are we 100% sure that nothing anywhere when its off might try and access the current_month attribute?